setArtifactVersion.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -e
+
+checkIfTagExists(){
+  version=$1
+  tagCount=$(curl -s https://api.github.com/repos/$GITHUB_REPOSITORY/tags | jq --arg tagName "$version"  '[.[] | select( .name == $tagName)] | length')
+  if [ $tagCount -gt 0 ]; then
+    echo "Error: Version $version already released. Please update your version in package/.appversion"
+    exit 1
+  fi
+}
+
+setVersionAsGithubEnv(){
+  version=$1
+  echo "Setting version $version"
+  echo "ARTIFACT_VERSION=$version" >> $GITHUB_ENV
+}
+
+case $GITHUB_REF in
+  refs/tags/*)
+      echo "Current action is for tag.."
```

Comment:
these echo logs are to indicate the action intent - such as `/tags` would create release version, `/heads/release-*` would create release candidate artifact version etc

Reply:
Yes Nouman.